src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -75,6 +76,7 @@ public TestWorkspace(ExportProvider exportProvider, string? workspaceKind = null
             _backgroundParser.Start();
 
             _metadataAsSourceFileService = exportProvider.GetExportedValues<IMetadataAsSourceFileService>().FirstOrDefault();
+            _testExtensionErrorHandler = exportProvider.GetExportedValue<TestExtensionErrorHandler>();
```

Comment:
❗️ This should move to `UseExportProviderAttribute`. `TestExtensionErrorHandler` is a MEF part, and that attribute contains the test cleanup code that applies to all tests using MEF, whether or not they use `TestWorkspace`.

Reply:
Not sure what you're suggesting here. Can you provide an example for me to use as reference here?

Reply:
I was being literal. Here is the MEF cleanup code:

https://github.com/dotnet/roslyn/blob/b2254719aa74c9c96bd3df1b9be96fb73727fec1/src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs#L80-L153

Reply:
This one will almost certainly operate in the same manner as `listenerProvider`, and occur _after_ that one in the `try` block to ensure we still throw if pending asynchronous work ends up triggering the failure.